sdk/src/types/query.ts

```diff
@@ -56,15 +56,25 @@ export type MergeFields<FieldList> = HasNestedFields<FieldList> extends never
 	: Extract<UnpackList<FieldList>, string> | MergeRelationalFields<FieldList>;
 
 /**
- * Query sort
- * TODO expand to relational sorting (same object notation as fields i guess)
+ * Query sort with support for nested relational fields
  */
-export type QuerySort<_Schema, Item> = UnpackList<Item> extends infer FlatItem
-	? {
-			[Field in keyof FlatItem]: Field | `-${Field & string}`;
-	  }[keyof FlatItem]
+export type QuerySort<Schema, Item> = UnpackList<Item> extends infer FlatItem
+	? readonly (keyof FlatItem | `-${keyof FlatItem & string}` | RelationalSort<Schema, FlatItem>)[]
 	: never;
 
+/**
+ * Handle relational field sorting using dot notation
+ */
+type RelationalSort<Schema, Item> = {
+	[Key in RelationalFields<Schema, Item>]: Extract<Item[Key], ItemType<Schema>> extends infer RelatedCollection
+		? RelatedCollection extends any[]
+			? HasManyToAnyRelation<RelatedCollection> extends never
+				? never // many-to-many or one-to-many should not be sortable
+				: never // many-to-any
```

Comment:
These are in fact sortable and should be represented here too.
![image](https://github.com/user-attachments/assets/0041f2ef-10f3-4eca-931b-1cf194481d45)